Detections/MultipleDataSources/BrutforceAttemptOnAzurePortalAndAWSConsolAtSameTime.yaml

```diff
@@ -0,0 +1,106 @@
+id: 1f40ed57-f54b-462f-906a-ac3a89cc90d4
+name: Cross-Cloud Credential Brute Force Attempt
+description: |
+  'This detection focuses on identifying potential cross-cloud brute force attempts involving Azure and AWS platforms. It monitors sign-in activities within the Azure Portal and AWS ConsoleLogins where brute force attempts are successful on both platforms in a synchronized manner.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: AWS
+    dataTypes:
+      - AWSCloudTrail
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - SigninLogs
+  - connectorId: BehaviorAnalytics
+    dataTypes:
+      - IdentityInfo
+  - connectorId: MicrosoftThreatProtection
+    dataTypes:
+      - SecurityAlert
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CredentialAccess
+relevantTechniques:
+  - T1110
+query: |
+  // Materialize a table named "Azure_Bruforce" containing Azure Portal sign-in logs within the last 1 day
+  let Azure_Bruforce =  materialize (
+      SigninLogs
+  // Filter sign-in logs related to the Azure Portal
+        | where AppDisplayName == "Azure Portal"
+  // Exclude entries with empty OriginalRequestId
+      | where isnotempty(OriginalRequestId)
+  // Limit the time range to the last 1 day
+      | where TimeGenerated >= ago(1d)
+  // Summarize various counts and sets based on brute force criteria
+      | summarize 
+          SuccessfulEvent = countif(ResultType == 0), 
+          FailedEvent = countif(ResultType != 0), 
+          totalAzureLoginEventId = dcount(OriginalRequestId), 
+          FailedEventsCount = dcountif(OriginalRequestId, ResultType != 0), 
+          SuccessfuleventsCount = dcountif(OriginalRequestId, ResultType == 0),
+          SetOfFailedevents = makeset(iff(ResultType != 0, OriginalRequestId, ""), 5), 
+          SetOfSuccessfulEvents = makeset(iff(ResultType == 0, OriginalRequestId, ""), 5) 
+  // Grouping by various attributes
+          by 
+          IPAddress, 
+          UserPrincipalName, 
+          bin(TimeGenerated, 1min), 
+          UserAgent,
+          ConditionalAccessStatus,
+          OperationName,
+          RiskDetail,
+          AuthenticationRequirement,
+          ClientAppUsed
+  // Filtering based on conditions for failed and successful events
+      | where FailedEventsCount >= 5

```

Comment:
Seems odd to have these a different between Azure and AWS - is there a reason for this?

Reply:
This approach follows the Auch process, where each platform has a distinct request completion time. I've derived these times from the average of potential failed counts within a minute, aiding in minimizing false positive events and reducing noise.